src/routes/player/PlayQuiz.tsx

```diff
@@ -0,0 +1,92 @@
+import { Link } from "react-router-dom";
+import DisplayQuestion from "../../components/player/DisplayQuestion";
+import { useEffect } from "react";
+import { socket } from "../../socket";
+import { WS_EVENTS } from "../../socket/events";
+import { useCountdown } from "usehooks-ts";
+import { state$ } from "../../state";
+import ShowAnswer from "../../components/player/ShowAnswer";
+import WaitingForOtherPlayers from "../../components/player/WaitingForOtherPlayers";
+import WaitingForQuestion from "../../components/player/WaitingForQuestion";
+import { observer } from "@legendapp/state/react";
+
+// 30 seconds
+const TIMER_DURATION = 30;
+
+const PlayQuiz = observer(() => {
+  // nice hook to simplify the timer logic, default interval: 1000 ms (1 second)
+  const [count, { startCountdown }] = useCountdown({
+    countStart: TIMER_DURATION,
+  });
+
+  // send recv question event after component is mounted
+  useEffect(() => {
+    // let server know client is ready to answer question
+    socket.emit(WS_EVENTS.RECV_QUESTION, state$.roomId.get());
```

Comment:
we should be emitting RECV_QUESTION after receiving the question from the host, not on component being mounted I believe.

Reply:
yes you are right, but in `JoinRoom` once the `NEW_QUESTION` event is received, we navigate to this route `game/:room_id` and then we send the `RECV_QUESTION` to signify that the client is ready (the relevant components for the quiz are mounted)